.github/workflows/linter_test.yaml

```diff
@@ -0,0 +1,102 @@
+name: Linter Tests
+
+# This workflow is triggered on pushes to the repository or external PRs.
+on: [push, pull_request]
+
+jobs:
+  flake8:
+    # We want to run on external PRs, but not on our own internal PRs as they'll be run
+    # by the push to the branch.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != 'facebookresearch/mmf'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout branch 🛎️
+      uses: actions/checkout@v2
+
+    - name: Setup Conda Environment
+      uses: goanpeca/setup-miniconda@v1
+      with:
+        activate-environment: mmf
+        python-version: 3.7
+        auto-update-conda: true
+        use-only-tar-bz2: true
+
+    - name: Install Dependencies
+      shell: bash -l {0}
+      run: |
+        conda activate mmf
+        python -m pip install --upgrade pip
+        pip install --progress-bar off flake8==3.7.9
+
+    - name: Run flake8
+      shell: bash -l {0}
+      run: |
+        conda activate mmf
+        flake8 --version
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F72,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings
+        flake8 . --count --exit-zero --max-complexity=18 --max-line-length=88 --statistics
+
```

Comment:
IIRC, there is a way to make this modular by refactoring and depending on workflows.